nodes/generator/script1_lite.py

```diff
@@ -324,6 +320,18 @@ def process(self):
             return
         self.process_script()
 
+    def get_node_dict(self):
+        self.n_id = node_id(self)
+        return self.node_dict.get(self.n_id)
+
+    def reset_node_dict(self):
+        self.node_dict[self.n_id] = {}
+
+    def set_node_dict_keys(self, **kwargs):
+        ND = self.node_dict.get(self.n_id)
+        for k, v in kwargs.items():
+            ND[k] = v
```

Comment:
What about using of standard Python dictionary API?
```suggestion
    @property
    def node_dict(self):
        self.n_id = node_id(self)
        return self._node_dict.get(self.n_id)
```

```Python
self.node_dict.clear()
self.node_dict['key'] = my_value
```

Reply:
Node_dict can use `defaultdict` if needed.

```Python
class Node:
    _node_dict = defaultdict(dict)
```

https://docs.python.org/3/library/collections.html#defaultdict-examples

Reply:
i like to make small changes first.

Reply:
what you propose there is in retrospect maybe preferable,  and i will address this later. it looks better, but for now i'm looking for a bug in the prop_search